backend/backend.go

```diff
@@ -122,3 +122,38 @@ func UnmarshalHistoryEvent(bytes []byte) (*HistoryEvent, error) {
 	}
 	return e, nil
 }
+
+func GetSubOrchestrationInstances(ctx context.Context, be Backend, iid api.InstanceID, isPurge bool) ([]api.InstanceID, error) {
```

Comment:
Does this need to be exported or can we make it private?

```suggestion
func getSubOrchestrationInstances(ctx context.Context, be Backend, iid api.InstanceID, isPurge bool) ([]api.InstanceID, error) {
```